src/Input/Select/Select.test.tsx

```diff
@@ -126,12 +131,126 @@ describe('when no results are found', () => {
 
 describe("when status = 'error'", () => {
   it('should show a red border', () => {
-    const { getByRole } = render(<Select status="error">{SelectChildren}</Select>);
+    const { getByRole } = render(
+      <Select status="error">{SelectChildren}</Select>
+    );
     const selectInput = getByRole('combobox');
     expect(selectInput).toHaveStyle(`border-color: ${PrimaryColor.glintsred}`);
   });
 });
 
+describe('when error is true', () => {
+  it('should show a red border', () => {
+    const { getByRole } = render(
+      <Select error={true}>{SelectChildren}</Select>
+    );
+    const selectInput = getByRole('combobox');
+    expect(selectInput).toHaveStyle(`border-color: ${PrimaryColor.glintsred}`);
+  });
+});
+
+describe('when error is a string', () => {
+  it('should show a red border', () => {
+    const { getByRole } = render(
+      <Select error="error">{SelectChildren}</Select>
+    );
+    const selectInput = getByRole('combobox');
+    expect(selectInput).toHaveStyle(`border-color: ${PrimaryColor.glintsred}`);
+  });
+});
+
+describe('when error is a component', () => {
+  it('should show a red border', () => {
+    const { getByRole } = render(
+      <Select error={<span>error</span>}>{SelectChildren}</Select>
+    );
+    const selectInput = getByRole('combobox');
+    expect(selectInput).toHaveStyle(`border-color: ${PrimaryColor.glintsred}`);
+  });
+});
+
+describe('when renderError is not given', () => {
```

Comment:
Do you think the test cases for both `when renderError is not given` and `when renderError is given` is necessary? My initial thought is that we can just test for `when renderError is given`. If not we risk duplicating test for every single prop passed into the component if we follow this convention. If we really want to test it, maybe we can put it in the `when it is rendered` block on line 74. Wdyt?

Reply:
I moved the tests, I think it's less confusing this way. I really want these tests because I want to test the rather complex interaction between `error` and `renderError`. No need to test combinations with other props.

Reply:
Okay got it!